handle None cases when empty strings are returned from the core

psifertex opened the pull request:
Prior to this commit (and the broken "fix" before it in [ac909374541f32697a065c43e7e12c5d6916f937](https://github.com/Vector35/binaryninja-api/commit/ac909374541f32697a065c43e7e12c5d6916f937#diff-6782036b11fde2c417fa4d886777898de32ff7edff2d3d0e6171614516127f3aR167-R168)), the following API failed:

`CoreVersionInfo("4.2.0")` 

This resolves the issue by checking if a string is empty before trying to decode it and failing.

psifertex committed:
handle None cases when empty strings are returned from the core and fix associated coreversion usage

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Vector35/binaryninja-api?pullRequest=7614) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Vector35/binaryninja-api?pullRequest=7614) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: fuzyll<br/>:x: psifertex<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Vector35/binaryninja-api?pullRequest=7614) it.</sub>

fuzyll committed:
Other changes related to handling null strings.

This fixes the incorrect type hint and the other instance of the
function. It also looks like type parsing could have the same issue, so
I've updated that as well.

fuzyll commented:
I've added some things on top of this, including unit tests. Unfortunately, that means we can't handle the PR at the API level anymore. So, I'm closing this one in favor of the new PR I've created that also includes this API branch.
